### PR TITLE
feat: Added h3 server handler to @ingest/h3.

### PR DESCRIPTION
**Pull request by michealroberts**
## Summary

Added h3 server handler to @ingest/h3.

h3 is the unjs project which is the foundation for the Nuxt 3 server `eventHandlers`.

This MR provides backwards compatibility between Nuxt and h3, and can also be used within a nitro server setup too.

Includes bumping h3 to the latest version (v1.8.1).

## Checklist

- [ ] Added a [docs PR](https://github.com/inngest/website) that references this PR
- [X] Added unit/integration tests
- [X] Added changesets if applicable

### CODE REVIEW

**Comment by michealroberts** [comment]
cc @pi0 Just pinging you here, because usage of the `getRequestURL` utility from h3 fails when used in place of the code in this MR ... it fails on this error:

```
 TypeError: Cannot read properties of undefined (reading 'encrypted')
```

Due to this line here: https://github.com/unjs/h3/blob/12ccb81432a62783fe951a8170976c8162bce44a/src/utils/request.ts#L129

Do you want me to add an issue to https://github.com/unjs/h3 ?

**Comment by pi0** [comment]
Hi. Haven’t tried locally but are you somehow mocking request and response objects passed to h3 habdler? 

Regardless feel free to directly open a pr for h3 to make req.connection?. Check optional

**Comment by michealroberts** [comment]
@pi0 Cool. MR here for you when ready: https://github.com/unjs/h3/pull/532

**Review by jpwilliams — CHANGES_REQUESTED** [review]
This is great! Thanks for the MR, @michealroberts!

One mini change to some tests and we can get this in. Even has a changeset already! 😍

File: packages/inngest/src/nuxt.test.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import * as NuxtHandler from "@local/nuxt";
+import * as NuxtHandler from "@local/h3";
```
[the remark this answers is not in the document]
Comment (michealroberts) [reply packages/inngest/src/nuxt.test.ts:1]: @jpwilliams Sure! :) Will ammend.

**Comment by jpwilliams** [comment]
@michealroberts I pushed a quick change to ensure handlers are still reported correctly for the benefit of the UI. It's a touch ugly, but this is changing soon anyway.

Could I grab an example for you for a typical short h3 setup? I'll add an example to the [Serving the Inngest API](https://www.inngest.com/docs/sdk/serve) docs. 🙂

**Comment by michealroberts** [comment]
> @michealroberts I pushed a quick change to ensure handlers are still reported correctly for the benefit of the UI. It's a touch ugly, but this is changing soon anyway.
> 
> Could I grab an example for you for a typical short h3 setup? I'll add an example to the [Serving the Inngest API](https://www.inngest.com/docs/sdk/serve) docs. 🙂

Cool, yeh I will add an example. It's exactly like Nuxt, but it can be a tad esoteric to know. This would also support nitro as well, which Nuxt is built on.

Kind of like turtles all the way down.

Nuxt 3 > Nitro > h3

**Comment by jpwilliams** [comment]
@michealroberts Just a quick snippet is fine 🙂 Would something like this be reasonable?

```ts
import { createApp, eventHandler } from "h3";
import { serve } from "inngest/h3";

const app = createApp();

app.use(
  "/api/inngest",
  eventHandler(serve(...)),
);
```

**Comment by michealroberts** [comment]
@jpwilliams Yaeh, exactly that!

Do you want the same again, but for nitro? https://nitro.unjs.io/

**Review by jpwilliams — APPROVED** [review]

**Pull request merged**